stream_alert/rule_processor/config.py

```diff
@@ -33,15 +35,29 @@ def load_config(conf_dir='conf/'):
     key denotes the name of the log type, and includes 'keys' used to match
     rules to log fields.
     """
-    conf_files = ('sources', 'logs', 'types', 'global')
+    conf_files = ('sources', 'logs', 'types', 'global', 'clusters')
     config = dict()
     for base_name in conf_files:
-        path = '{}.json'.format(os.path.join(conf_dir, base_name))
-        with open(path) as data:
-            try:
-                config[base_name] = json.load(data, object_pairs_hook=OrderedDict)
-            except ValueError:
-                raise ConfigError('Invalid JSON format for {}.json'.format(base_name))
+        if base_name == 'clusters':
+            # Load current cluster config into memory for threat intel
+            path = os.path.join(conf_dir, base_name)
+            config['clusters'] = {}
+            if CLUSTER:
+                cluster_conf_path = '{}.json'.format(os.path.join(path, CLUSTER))
+                with open(cluster_conf_path) as data:
+                    try:
+                        config['clusters'][CLUSTER] = json.load(data)
+                    except ValueError:
+                        raise ConfigError('Invalid JSON format for {}'.format(cluster_conf_path))
+        else:
+            path = '{}.json'.format(os.path.join(conf_dir, base_name))
+            with open(path) as data:
+                try:
+                    # we use object_pairs_hook=OrderdDict to preserve schema order
```

Comment:
Can this condense into a single line? or is it above limits

Reply:
it will be above limits (104 chars) if condense into a single line.

Reply:
Just so I'm clear, the requirement for the objet pairs hook is only really for the logs.json file, correct? We could (theoretically) check if that's the file being loaded and only use that keyword arg for that specific file, instead of all non-cluster files.

EDIT: probably not a change that needs to happen now, just a thought